src/Entity/Routing/EntityTypeHtmlRouteProvider.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * @file
+ *   Contains \Drupal\content_entity_base\Entity\Routing\EntityTypeHtmlRouteProvider.
+ */
+
+namespace Drupal\content_entity_base\Entity\Routing;
+
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\Routing\AdminHtmlRouteProvider;
+use Symfony\Component\Routing\Route;
+
+/**
+ * Provides routes for custom content entities bundles.
+ *
+ * @see Drupal\Core\Entity\Routing\AdminHtmlRouteProvider
+ * @see Drupal\Core\Entity\Routing\DefaultHtmlRouteProvider
+ */
+class EntityTypeHtmlRouteProvider extends AdminHtmlRouteProvider {
+  /**
+   * {@inheritdoc}
+   */
+  public function getRoutes(EntityTypeInterface $entity_type) {
+    $routes = parent::getRoutes($entity_type);
+
+    $entity_type_id = $entity_type->id();
+
+    if ($collection_route = $this->getCollectionRoute($entity_type)) {
+      $routes->add("entity.{$entity_type_id}.collection", $collection_route);
+    }
+
+    if ($add_form_route = $this->getAddFormRoute($entity_type)) {
```

Comment:
Oh well, the CruiUIRouteProvider actually has the collection, so i'm not sure why this one is actually needed